ocean_lib/ocean/ocean_assets.py

```diff
@@ -141,24 +141,77 @@ def build_data_tokens_list(
         self, services: list, deployed_erc20_tokens: list
     ) -> list:
         data_tokens = []
+        # (1-n) service per datatoken, 1 datatoken per service
         for erc20_token in deployed_erc20_tokens:
             for service in services:
-                data_token = {
-                    "address": erc20_token.address,
-                    "name": erc20_token.contract.caller.name(),
-                    "symbol": erc20_token.symbol(),
-                    "serviceId": service.id,
-                }
-                data_tokens.append(data_token)
+                if service.data_token == erc20_token.address:
+                    data_tokens.append(
+                        {
+                            "address": erc20_token.address,
+                            "name": erc20_token.contract.caller.name(),
+                            "symbol": erc20_token.symbol(),
+                            "serviceId": service.id,
+                        }
+                    )
         return data_tokens
 
+    @staticmethod
+    def process_ddo(
+        asset: Asset,
+        provider_uri: str,
+        encrypt_flag: Optional[bool] = False,
+        compress_flag: Optional[bool] = False,
+    ):
+        # Process the DDO
+        asset_dict = asset.as_dictionary()
+        ddo_string = json.dumps(asset_dict, separators=(",", ":"))
+        ddo_bytes = ddo_string.encode("utf-8")
+        ddo_hash = create_checksum(ddo_string)
+
+        # Plain asset
+        if not encrypt_flag and not compress_flag:
```

Comment:
I also think we could use some "return" statements here, instead of doing a huge `elif` thing. This was previously an `elif` because it was part of a larger logic, but now as a function it *screams* "force returns". E.g. if the asset is plain you can return it directly...